misc/python/materialize/parallel_workload/database.py

```diff
@@ -175,7 +200,64 @@ def create(self, exe: Executor) -> None:
         exe.execute(query)
 
 
-class Role(DBObject):
+class WebhookColumn(Column):
+    def __init__(
+        self, name: str, data_type: type[DataType], nullable: bool, db_object: DBObject
+    ):
+        self._name = name
+        self.data_type = data_type
+        self.nullable = nullable
+        self.db_object = db_object
+
+    def __str__(self) -> str:
+        return f"{self.db_object}.{self._name}"
+
+    def name(self) -> str:
+        return self._name
+
+
+class WebhookSource(DBObject):
+    source_id: int
+    rename: int
+    cluster: "Cluster"
+    body_format: BodyFormat
+    include_headers: list[str]
+
+    def __init__(self, source_id: int, cluster: "Cluster", rng: random.Random):
+        self.source_id = source_id
+        self.cluster = cluster
+        self.rename = 0
+        self.body_format = rng.choice([e for e in BodyFormat])
+        self.include_headers = []
+        for i in range(rng.randint(0, MAX_INCLUDE_HEADERS)):
+            self.include_headers.append(f"ih{i}")
+        self.columns = [
+            WebhookColumn(
+                "body",
+                self.body_format.to_data_type(),
+                False,
+                self,
+            )
+        ] + [
+            WebhookColumn(include_header, Text, True, self)
+            for include_header in self.include_headers
+        ]
+        # TODO: INCLUDE HEADERS (as map)
+        # TODO: CHECK WITH
+
+    def __str__(self) -> str:
+        if self.rename:
+            return f"wh{self.source_id}_{self.rename}"
+        return f"wh{self.source_id}"
+
+    def create(self, exe: Executor) -> None:
+        query = f"CREATE SOURCE {self} IN CLUSTER {self.cluster} FROM WEBHOOK BODY FORMAT {self.body_format.name}"
```

Comment:
can you also do `INCLUDE HEADERS` and then
 ```
 CHECK (
      WITH (
        HEADERS,
        BODY AS request_body) LENGHTH(request_body) > 0 )
```

That is, exercise a few more code paths in the webhook sources while still having all the incoming POSTs pass the `CHECK` check.

Reply:
Oh I see you are including some headers already below, so I guess only do an always-true `CHECK` clause then.